src/store.rs

```diff
@@ -281,4 +286,36 @@ mod tests {
 
         assert_eq!(updated_ticket.status, Status::Done)
     }
+
+    #[test]
+    fn add_comment_to_ticket() {
+        //arrange
+        let mut ticket_store = TicketStore::new();
+        let ticket = generate_and_persist_ticket(&mut ticket_store);
+        let comment = Comment::new("Test Comment".to_string()).unwrap();
+        let expected = Comment::new("Test Comment".to_string()).unwrap();
+
+        //act
+        let result = ticket_store.add_comment_to_ticket(ticket.id, comment);
+        //assert
+        assert!(result.is_some());
+        let ticket = ticket_store.get(ticket.id).unwrap();
+        assert_eq!(ticket.comments.len(), 1);
+        assert_eq!(ticket.comments[0], expected);
```

Comment:
You could rephrase this as `assert_eq!(ticket.comments, vec![comment])`

Reply:
I have updated to rephrase as suggested. I couldn't do it exactly like that as comment does not - and from my understanding cannot - implement Copy. I have cloned comment into and expected comment and rephrased to `assert_eq!(ticket.comments, vec![expected_comment]`

Reply:
I have also merged Master into this branch and resolved issues. I've added Serialization to comment too.